xdsl/dialects/builtin.py

```diff
@@ -1704,17 +1705,20 @@ def get_element_type(self) -> _UnrankedMemrefTypeElems:
     VectorType[AttributeCovT] | TensorType[AttributeCovT] | MemRefType[AttributeCovT]
 )
 
+AnyDenseElement: TypeAlias = IntegerType | IndexType | AnyFloat
+DenseElementT = TypeVar("DenseElementT", bound=AnyDenseElement, covariant=True)
+_DenseElementT = TypeVar("_DenseElementT", bound=AnyDenseElement)
```

Comment:
Is this the equivalent of sized in MLIR? Would it make sense to add this independently of the other changes?

Reply:
I doubt it is as `IndexType` is not sized right? (Also `IndexType` should be forbidden from `DenseIntOrFPElementsAttr` but that's another matter)

Reply:
I think it's quite relevant, and we should forbid it indeed, I thought we already made that change recently.

Reply:
Ah it was DenseArrayBase #3258